ui/src/frontend/rpc_http_dialog.ts

```diff
@@ -304,36 +310,257 @@ enum PreloadedDialogResult {
+                ],
+              );
+            });
+
+            const newInstanceSelected = selectedInstanceId === null;
+
+            const newClasses = [
+              'pf-new-row',
+              newInstanceSelected ? 'pf-new-row--selected' : '',
+            ].join(' ');
+
+            elements.push(
+              m(CardStack, [
+                ...rows,
+                m(
+                  'div',
+                  {
+                    key: 'new-instance-row',
+                    role: 'option',
+                    tabindex: 0,
+                    onclick: () => {
+                      selectedInstanceId = null;
+                      redrawModal();
+                    },
+                    onkeypress: (e: KeyboardEvent) => {
+                      if (e.key === 'Enter' || e.key === ' ') {
+                        e.preventDefault();
+                        selectedInstanceId = null;
+                        redrawModal();
+                      }
+                    },
+                    class: newClasses,
+                  },
+                  [
+                    m('div', m('strong', 'New instance …')),
+                    newInstanceSelected &&
+                      m(
+                        'div',
+                        {
+                          class: 'pf-selected-inline',
+                        },
+                        'Selected',
+                      ),
+                  ],
+                ),
+              ]),
+            );
+          } else {
+            elements.push(
+              m(
+                'p',
+                {class: 'pf-no-sessions'},
+                `There are no current active sessions on ${HttpRpcEngine.hostAndPort}.`,
+              ),
+            );
+
+            // ensure default selection is "New instance" if nothing exists
+            selectedInstanceId = null;
+          }
+
+          return elements;
         },
-      },
-    ],
+        buttons: [
+          {
+            text: 'Yes, Attach to external RPC',
+            primary: true,
+            action: () => {
+              if (selectedInstanceId !== null) {
+                AppImpl.instance.httpRpc.selectedTraceProcessorId =
+                  selectedInstanceId;
+                closeModal();
+                resolve(PreloadedDialogResult.UseRpcWithPreloadedTrace);
+                return;
+              }
+
+              closeModal();
+              resolve(PreloadedDialogResult.UseRpc);
+            },
+          },
+          {
+            text: 'Use built-in WASM',
+            action: () => {
+              closeModal();
+              resolve(PreloadedDialogResult.UseWasm);
+            },
+          },
+        ],
+      });
+
+      function redrawModal() {
+        try {
+          const mWithRedraw = m as {redraw?: () => void};
+          if (typeof mWithRedraw.redraw === 'function') {
+            mWithRedraw.redraw();
+          }
+        } catch (e) {}
+      }
```

Comment:
You really shouldn't need this. For a start, m.redraw() already exists and is well defined in typescript. Also, the autoredraw system will automatically redraw after running event handlers registered via mithril